File: src/main/java/com/vaadin/demo/component/emailfield/EmailFieldStyles.java
package com.vaadin.demo.component.emailfield;

import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.textfield.EmailField;
import com.vaadin.flow.component.textfield.TextFieldVariant;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("email-field-styles")
public class EmailFieldStyles extends Div {

    public EmailFieldStyles() {
        // tag::snippet[]
        EmailField field = new EmailField();
        field.addThemeVariants(TextFieldVariant.SMALL,
                TextFieldVariant.ALIGN_RIGHT, TextFieldVariant.HELPER_ABOVE);
        // end::snippet[]
        field.setLabel("Label");
        field.setHelperText("Helper text");
        field.setValue("foo@bar.com");

        add(field);
    }

    public static class Exporter extends DemoExporter<EmailFieldStyles> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/emailfield/email-field-styles.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/email-field';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('email-field-styles')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-email-field
        theme="align-right small helper-above-field"
        label="Label"
        helper-text="Helper text"
        value="foo@bar.com"
      ></vaadin-email-field>
      <!-- end::snippet[] -->
    `;
  }
}
